__tests__/app.test.js

```diff
@@ -92,59 +92,117 @@ describe("GET /api/articles", () => {
+        const commentArray = JSON.parse(body);
+        const expected = {
+          comment_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+        };
+        commentArray.forEach((comment) => {
+          expect(comment).toMatchObject(expected);
+        });
         expect(Array.isArray(commentArray)).toBe(true);
-      })
+      });
   });
   test("404:responds with an error, this article_id does not exist yet", () => {
     return request(app)
-      .get('/api/articles/100/comments')
+      .get("/api/articles/100/comments")
       .expect(404)
-        .then((response) => {
-          expect(response.body.msg).toBe("this article does not exist yet");
-        });
-    });
+      .then((response) => {
+        expect(response.body.msg).toBe("this article id does not exist yet");
+      });
+  });
+  test("200:valid input, with a recognised id, but reponds with empty array and appropiate message", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.msg).toBe(
+          "this article does not have any comments"
+        );
+        expect(response.body.comments).toEqual([]);
+      });
+  });
 });
-test("404:responds with an error, this article_id does not have any comments", () => {
-  return request(app)
-    .get('/api/articles/2/comments')
-    .expect(404)
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201:responds with an object, a comment with properties of username and body", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "60% of the time it works every time",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(newComment)
+      .expect(201)
       .then((response) => {
-        expect(response.body.msg).toBe("this article does not have any comments");
+        expect(response.body).toEqual(
+          expect.objectContaining({
+            username: newComment.username,
+            body: newComment.body,
+          })
+        );
       });
   });
-
+test("404:responds with an error, this article_id does not exist yet", () => {
+  return request(app)
+    .post("/api/articles/100/comments")
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe("this article id does not exist yet");
+    });
+});
+test("404:responds with an error, this username does not exist yet", () => {
+  const newComment = {
+    username: "RonBurgundy",
+    body: "60% of the time it works every time",
+  };
+  return request(app)
+    .post("/api/articles/2/comments")
+    .send(newComment)
+    .expect(404)
+    .then((response) => {
+      expect(response.body.msg).toBe("Sorry, but this username doesn't exist.");
+    });
```

Comment:
This is a really good set of tests for 404 Not Found errors, where our inputs are valid! 🥇 But with POST and PATCH, we need information from our user. What happens if they provide bad information?
- An `article_id` that is a string for example?
- A `newComment` that is missing one or both keys? Or contains objects as values?